lib/mongoid/traversable.rb

```diff
@@ -35,6 +35,11 @@ def discriminator_key=(value)
           class << self
             delegate :discriminator_key, to: ::Mongoid
           end
+        end 
+
+        unless fields.has_key?(self.discriminator_key) || descendants.length == 0
```

Comment:
Can this condition be reworded using `if` please?

Reply:
And you can use `descendants.empty?` instead of `descendants.length == 0`.

Reply:
Also can this conditional have a comment above it explaining what it is for (i.e. adding the new field for the new discriminator_key to already instantiated classes that have descendants, and thus that have had the previous discriminator key field added to them already).

Reply:
done!